Amazon.IonDotnet/Internals/Binary/RawBinaryReader.cs

```diff
@@ -700,23 +715,21 @@ protected Timestamp ReadTimeStamp(int length)
                             {
                                 // now we read in our actual "milliseconds since the epoch"
                                 frac = ReadDecimal(_localRemaining);
+                                // if fractional second does not exist, it will be set to 0. However, we can have 0.0 or 0.00
+                                // which are valid and different from 0. So anything other than 0 is a fractional second. 
+                                if (!frac.ToString().Equals("0"))
```

Comment:
Is converting to string the only way? Can we instead just check whether the value is 0 and has an exponent < 0?

Reply:
Unfortunately as far as I read, there is not a better way.